packages/vite/src/node/plugins/dynamicImportVars.ts

```diff
@@ -174,8 +174,9 @@ export function dynamicImportVarsPlugin(config: ResolvedConfig): Plugin {
   })
 
   if (
-    config.experimental.enableNativePlugin === true &&
-    config.command === 'build'
+    config.command === 'build' &&
+    (config.experimental.enableNativePlugin === true ||
+      config.experimental.enableNativePlugin === 'v1')
```

Comment:
Maybe it's easier if we have `ResolvedConfig.nativePluginEnabledLevel: number` so that we can check it like `config.nativePluginEnabledLevel >= 1`? (mapping `resolver` to `0` and `undefined` to `-1`)